vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/steps/integration/ResourceCheckSteps.java

```diff
@@ -216,9 +217,20 @@ private String getCssSelector(Element element)
     private URI resolveUri(String uri) throws URISyntaxException
     {
         URI uriToCheck = new URI(uri);
+
         if (isNotAbsolute(uriToCheck))
         {
             URI mainApplicationPageUrl = webApplicationConfiguration.getMainApplicationPageUrlUnsafely();
+            if (uri.length() > 1 && uri.charAt(1) == '/')
+            {
+                String scheme = Optional.ofNullable(mainApplicationPageUrl)
+                                        .map(URI::getScheme)
+                                        .orElseThrow(() -> new IllegalStateException(
+                                                "Unable to evaluate protocol for main application page url: "
+                                                        + mainApplicationPageUrl));
+                return new URIBuilder(uri).setScheme(scheme).build();
+            }
+
             if (mainApplicationPageUrl != null)
             {
                 return buildNewUrl(mainApplicationPageUrl, uri);
```

Comment:
```java
    private URI resolveUri(String uri) throws URISyntaxException
    {
        URI uriToCheck = new URI(uri);

        if (isNotAbsolute(uriToCheck))
        {
            URI mainApplicationPageUrl = webApplicationConfiguration.getMainApplicationPageUrlUnsafely();

            if (mainApplicationPageUrl != null)
            {
                if (uri.length() <= 1 || uri.charAt(1) != '/')
                {
                    return buildNewUrl(mainApplicationPageUrl, uri);
                }
                String scheme = mainApplicationPageUrl.getScheme();
                if (scheme != null)
                {
                    return new URIBuilder(uri).setScheme(scheme).build();
                }
            }
        }
        return uriToCheck;
    }
```
?

Reply:
But it misses the point, I want to throw an exception if we got // URL and we can't determine the scheme, in your code the request to // will be still sent

Reply:
I'll take it but it needs to be reworked on exception in the future